service/controller/app/v1/resource/chart/desired.go

```diff
@@ -38,34 +38,35 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 			APIVersion: chartAPIVersion,
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        cr.Spec.Name,
+			Name:        cr.GetName(),
 			Namespace:   r.chartNamespace,
 			Labels:      processLabels(r.projectName, cr.ObjectMeta.Labels),
 			Annotations: cr.ObjectMeta.Annotations,
 		},
 		Spec: v1alpha1.ChartSpec{
-			Name:      cr.ObjectMeta.Name,
-			Namespace: cr.Spec.Namespace,
-			Config: v1alpha1.ChartSpecConfig{
-				ConfigMap: v1alpha1.ChartSpecConfigConfigMap{
-					Name:      key.ConfigMapName(cr),
-					Namespace: key.ConfigMapNamespace(cr),
-				},
-				Secret: v1alpha1.ChartSpecConfigSecret{
-					Name:      key.SecretName(cr),
-					Namespace: key.SecretNamespace(cr),
-				},
-			},
+			Name:       cr.ObjectMeta.Name,
+			Namespace:  key.Namespace(cr),
 			TarballURL: tarballURL,
 		},
 	}
 
+	if key.AppConfigMapName(cr) != "" || appcatalogkey.ConfigMapName(cc.AppCatalog) != "" {
```

Comment:
doesn't it have to be `&&`?

Reply:
No an OR is fine here. The app CR has 3 levels of config (catalog, app and user). But all 3 levels are optional.

The user support will come later because its not needed for VOO.